server/routerlicious/packages/lambdas/src/utils/messageGenerator.ts

```diff
@@ -48,3 +48,30 @@ export const createRoomLeaveMessage = (clientId: string): ISignalMessage => ({
 		content: clientId,
 	}),
 });
+
+export const IRuntimeSignalMessageBody = {
+	/**
+	 * Indicates that the data associated with an edit is or must be a `boolean`.
+	 */
+	content: "RuntimeMessage",
+} as const;
+
+/**
+ * Mirrors ISignalEnvelope from runtime definitions, for signals that come from an external
+ * caller (not sent by a client (so no 'clientSignalSequenceNumber') and are always addressed
+ * to the Container (so no 'address').
+ */
+export interface IRuntimeSignalEnvelope {
+	contents: {
+		type: string;
+		content: any;
+	};
+}
+
+export const createRuntimeMessage = (signalContent: IRuntimeSignalEnvelope): ISignalMessage => ({
+	clientId: null,
```

Comment:
Just to help me understand this change better: is `clientId` set as `null` here because the message is being sent from a service or, more generally, something other than a client?

Reply:
Is it possible that there are different external callers? in that case, do we have a need to tell them apart? In other words: having `null` as the `clientId` might make it not possible to distinguish messages coming from those different external callers

Reply:
> Is clientId set as null here because the message is being sent from a service or, more generally, something other than a client?

Something other than client.

> Is it possible that there are different external callers? in that case, do we have a need to tell them apart? In other words: having null as the clientId might make it not possible to distinguish messages coming from those different external callers

The way FluidFramework is set up right now, we need to have clientId to be null to route it the way we need it to. In a future redesign we can change that. For now we have to stick to this. If there's a need to distinguish, that can be in the signal content body and the app can dig in and figure it out.

Reply:
Nit: It would probably be worth adding some of this context as an inline comment here.

Reply:
@hedasilv I actually followed the code path in FF and realized that having clientId as a string would still work (it didn't originally). However, this value is populated on the server side, and there isn't really a way for the server to put in information that allows the app to act on it. So in this case, the answer to your original question is actually:

> is clientId set as null here because the message is being sent from a service 

The message is being sent from a service.